src/main/java/meteordevelopment/meteorclient/utils/player/DamageUtils.java

```diff
@@ -146,31 +137,34 @@ public static double bedDamage(LivingEntity player, Vec3d bed) {
         // Multiply damage by difficulty
         damage = getDamageForDifficulty(damage);
 
+        // Reduce by armour
+        damage = DamageUtil.getDamageLeft((float) damage, (float) getArmor(player), (float) EntityAttributeManager.getAttributeValue(player, EntityAttributes.GENERIC_ARMOR_TOUGHNESS));
+
         // Reduce by resistance
         damage = resistanceReduction(player, damage);
 
-        // Reduce by armour
-        damage = DamageUtil.getDamageLeft((float) damage, (float) player.getArmor(), (float) player.getAttributeInstance(EntityAttributes.GENERIC_ARMOR_TOUGHNESS).getValue());
-
         // Reduce by enchants
-        ((IExplosion) explosion).set(bed, 5, true);
-        damage = blastProtReduction(player, damage, explosion);
+        damage = blastProtReduction(player, damage);
 
-        if (damage < 0) damage = 0;
-        return damage;
+        return Math.max(damage, 0);
     }
 
     // Anchor damage
 
     public static double anchorDamage(LivingEntity player, Vec3d anchor) {
-        mc.world.removeBlock(BlockPos.ofFloored(anchor), false);
+        BlockPos anchorPos = BlockPos.ofFloored(anchor);
+        mc.world.removeBlock(anchorPos, false);
         double damage = bedDamage(player, anchor);
-        mc.world.setBlockState(BlockPos.ofFloored(anchor), Blocks.RESPAWN_ANCHOR.getDefaultState());
+        mc.world.setBlockState(anchorPos, Blocks.RESPAWN_ANCHOR.getDefaultState());
```

Comment:
The block state shouldn't be set to the default state, because in case the damage calcs get ran on an anchor that has a charge and it doesn't get blown up, it will remove the charge from the client. The block state should rather be cached and reused afterwards.

Reply:
I am already aware of this, I'm wondering if there's a way to fix this that doesn't require modifying the actual world instead